### ISSUE TITLE
files.associations  does not work

### ISSUE THREAD

**Issue by akosyakov** [theia-ide/theia#4716]
```
    "files.associations": {
        "*.go": "typescript"
    }
````

An editor for `*.go` languages should prefer typescript language, it does not.

**Comment by jonnyhork** [comment]
Hey @akosyakov, I've been sort of following along with this issue but I was hoping you could clarify some thing for me. If I understand the current state, it seems that the `files.associations` setting was added by #4713 but it is not yet functional after #4721? I'm curious because don't see the `files.associations` option in setting at all when I spin up the lates version of Theia `0.5.0-next.9043bc1d`. 
Do you have any idea when this implementation might be functional?

Thanks for the help on this, I appreciate it!

**Comment by akosyakov** [comment]
It is tracked https://github.com/theia-ide/theia/issues/4581. @tolusha could you help to answer please

**Comment by tolusha** [comment]
@jonnyhork 
As @akosyakov  mentioned an implementation is tracked in https://github.com/theia-ide/theia/issues/4581
My initial plan was to fix extension loading https://github.com/theia-ide/theia/pull/4721
For the time being I postponed the issue https://github.com/theia-ide/theia/issues/4581

**Comment by jonnyhork** [comment]
Ok thanks, I'll keep an eye on #4581.

>For the time being I postponed the issue #4581

Does this mean there is currently no plan to continue work on the implementation at the moment? 
I'm asking just so I can have a better understanding of the timeline for custom file associations and plan accordingly.

**Comment by tsmaeder** [comment]
We'll get back to it in our next sprint (starting week after next)

**Comment by jonnyhork** [comment]
@tsmaeder Hi, I just wanted to check in about this issue. I see that file.associations was added to [this sprint](https://github.com/eclipse/che/issues/13074) but I am hoping you could speak to and estimated time for implementation. This would help me decide wether to use a workaround solution depending on the timeline. Thanks again for your help!

**Comment by tolusha** [comment]
@jonnyhork 
I would suggest using workaround for now.

**Comment by akosyakov** [comment]
fixed by https://github.com/theia-ide/theia/pull/4973
